### PR TITLE
CB-5114 Edit SSH key in environment

### PR DESCRIPTION
**Pull request by topolyai5**

### CODE REVIEW

File: cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPublicKeyConnector.java [hunk 1]
```diff
@@ -64,6 +62,21 @@ public void unregister(PublicKeyUnregisterRequest request) {
...
+    @Override
+    public boolean exists(PublicKeyDescribeRequest request) {
+        LOGGER.debug("Describe public key {} in {} region on AWS", request.getPublicKeyId(), request.getRegion());
+        AwsCredentialView awsCredential = new AwsCredentialView(request.getCredential());
+        try {
+            AmazonEC2Client client = awsClient.createAccess(awsCredential, request.getRegion());
+            return exists(client, request.getPublicKeyId());
+        } catch (Exception e) {
+            String errorMessage = String.format("Failed to describe public key [roleArn:'%s', region: '%s'], detailed message: %s",
+                    awsCredential.getRoleArn(), request.getRegion(), e.getMessage());
```
Comment (doktoric) [inline cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsPublicKeyConnector.java]: if the credential is access + secretkey then what happens?

File: environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java [hunk 2]
```diff
@@ -135,10 +140,28 @@ private void editDescriptionIfChanged(Environment environment, EnvironmentEditDt
...
         AuthenticationDto authenticationDto = editDto.getAuthentication();
         if (authenticationDto != null) {
+            EnvironmentValidatorService validatorService = environmentService.getValidatorService();
+            ValidationResult validationResult = validatorService.validateAuthenticationModification(editDto, environment);
+            if (validationResult.hasError()) {
+                throw new BadRequestException(validationResult.getFormattedErrors());
+            }
+            String oldSshKeyIdForDeletion = null;
+            String oldSshKeyId = environment.getAuthentication().getPublicKeyId();
+            if (environment.getAuthentication().isManagedKey()) {
```
Comment (doktoric) [inline environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java]: how about if we ccreate a method in environement and then calling like environment.isEnvironmentMAnagedKEy() ?

File: environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java [hunk 3]
```diff
@@ -113,6 +118,28 @@ public ValidationResult validateSecurityGroups(EnvironmentEditDto editDto, Envir
 
     }
 
+    public ValidationResult validateAuthenticationModification(EnvironmentEditDto editDto, Environment environment) {
+        ValidationResult.ValidationResultBuilder validationResultBuilder = ValidationResult.builder();
+
+        AuthenticationDto authenticationDto = editDto.getAuthentication();
+        String publicKeyId = authenticationDto.getPublicKeyId();
+        if (!environment.getCloudPlatform().equals(CloudPlatform.AWS.name()) && StringUtils.isNotEmpty(publicKeyId)) {
```
Comment (doktoric) [inline environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java]: can we somehow do this more dynamic so the the cloudconnector giving us this info ?

File: environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java [hunk 4]
```diff
@@ -113,6 +118,28 @@ public ValidationResult validateSecurityGroups(EnvironmentEditDto editDto, Envir
...
+            validationResultBuilder.error("The change of public key id is not supported on " + environment.getCloudPlatform());
+        } else {
+            String publicKey = authenticationDto.getPublicKey();
+            if (StringUtils.isNotEmpty(publicKeyId) && StringUtils.isNotEmpty(publicKey)) {
+                validationResultBuilder.error("You should define either public-key or public-key-id only");
+            }
+            if (StringUtils.isEmpty(publicKeyId) && StringUtils.isEmpty(publicKey)) {
+                validationResultBuilder.error("You should define public-key or public-key-id");
```
Comment (doktoric) [inline environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java]: I think it should reference for the json name

File: environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java (hunk 2 above)
Comment (topolyai5) [reply environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentModificationService.java]: the method name does not describe well the functionality.

**Pull request merged**